backend/src/requests/user-content-request/user-content-request.service.ts

```diff
@@ -111,37 +142,196 @@ export class UserContentRequestService {
    * @throws NotFoundException if no question is found with that id
    * */
   async getAnswersOfQuestion(id: string, sortCriteria: QueryParameters) {
+    // check question does exist
     const question = await this.userContentService.getQuestion(id);
     if (question == null)
       throw new NotFoundException('No question found with this id.');
 
+    // fetch answers
     const rawAnswers = await this.userContentService.getAnswersOfGroupID(
       question?.userContent?.groupID,
       sortCriteria,
     );
-    if (rawAnswers == null)
+    if (rawAnswers == null) {
       throw new InternalServerErrorException(
         'There was an error fetching the answers.',
       );
-
-    const answers = [];
-
+    }
+    // change results to openAPI schema
+    const answers: object[] = [];
     for (const answer of rawAnswers) {
       //@ts-ignore
-      answers.push(this.getUserContent(answer.userContentID, Type.ANSWER));
+      answers.push(
+        //@ts-ignore
+        await this.getUserContent(answer.userContentID, Type.ANSWER),
+      );
     }
 
     return answers ?? [];
   }
 
+  /**
+   * Wrapper for creation of Questions,
+   * does Input validation and error handling
+   * Tags are created if they do not exist in the database at the time the question is created
+   * @param data
+   * @param userId
+   * @throws UnprocessableEntityException
+   * @throws NotAcceptableException
+   */
+  async createQuestionWrapper(
+    data: CreateQuestion,
+    userId: string,
+  ): Promise<object> {
+    const forbiddenWords: string[] =
+      await this.blacklistService.getBlacklistArray(); // TODO buffer
+
+    // check basic data is present
+    if (data == null || data.content == null || data.title == null) {
+      throw new UnprocessableEntityException('Payload is not sufficient!');
+    }
+    const userExist: boolean = await this.userService.userIdExists(userId);
+    if (!userExist) {
+      throw new UnauthorizedException();
+    }
+
+    // check text for restricted words
+    if (
+      this.blacklistService.checkTextWithBlacklist(
+        data.title,
+        forbiddenWords,
+      ) ||
+      this.blacklistService.checkTextWithBlacklist(data.content, forbiddenWords)
```

Comment:
concat title with content, so that the function checkTextWithBlacklist only has to be called once